### ISSUE TITLE
Wrong scale computed in tile mode

### ISSUE THREAD

**Issue by mikitex70** [MapServer/MapServer#6912]
## Expected behavior and actual behavior.

Having a layer with `MAXSCALEDENOM` (whatever values are used) and requesting tiles (cgi request with `mode=tile&tilemode=gmap`, but I think the same is with `tilemode=ve`) the layer is never drawn.
Enabling logging show a message like:
```
Skipping layer (fogli) because LAYER.MAXSCALE is too small for this MAP scale
```

Debugging shows that, for example, for a zoom level `13` the computed scale was approximately `8025436685` (too large) instead of about `72224`.

It seems that in `maptile.c`, the function `msTileSetExtent` is missing a `map->units = MS_METERS;` since the calculated extent is now in meters.

Applied the following patch and now seems to work correctly:
```patch
--- a/maptile.c	2023-07-03 15:35:40.032816983 +0200
+++ b/maptile.c	2023-07-03 16:02:17.224862070 +0200
@@ -495,6 +495,8 @@
     msDebug( "msTileSetExtent (%f, %f) (%f, %f)\n", map->extent.minx, map->extent.miny, map->extent.maxx, map->extent.maxy);
   }
 
+  map->units = MS_METERS;  // now the units are meters
+
   return MS_SUCCESS;
 #else
   msSetError(MS_CGIERR, "Tile API is not available.", "msTileSetExtent()");
```

## Steps to reproduce the problem.

Execute the `test.sh` included in the attached zip file in  an environment with Docker (used only to build a test environment).
Or execute a command like:
```shell
mapserv -nh "QUERY_STRING=map=test.map&mode=tile&tilemode=gmap&tile=X+Y+Z&layers=testLayer" > out.jpg
```
using whatever map and tile coordinates you want and with the `testLayer` having the `MAXSCALEDENOM` set to any value.

## Attach simple test case
[tilescale-test.zip](https://github.com/MapServer/MapServer/files/11947725/tilescale-test.zip)

Run the `test.sh` script; it requires Docker to build a test envirtonment, but commenting the `docker build` and the `alias` commands it should work with the system installation of the `mapserv`.

## Operating system

Tested on CentOS 6.7 64 bit and Ubuntu 22.04 64 bit.

## MapServer version and installation method

MapServer 7.6.4 from Ubuntu repository.
MapServer version 7.6.5, self-compiled to enable the Oracle driver.

Not tested but seems that MapServer 8 is also affected by this issue.

**Comment by mikitex70** [comment]
Ok, I will fork the repository and apply the patch, but in which branch? `main`, `branch-7-6` or `branch-8-0`?

**Comment by jmckenna** [comment]
thanks @mikitex70, `main` is preferred.  Maybe your pull request can also expand the existing `mode_tile.map` test in msautotest.
- see https://github.com/MapServer/MapServer/blob/main/msautotest/misc/mode_tile.map
- for more info on msautotest in general see https://mapserver.org/development/tests/autotest.html

**Comment by jmckenna** [comment]
Good point, fixed through #6925.  Thanks again @mikitex70

**Comment by mikitex70** [comment]
Yes, the issue can be closed.
Thanks to all of you.
